internal/pkg/vm/internal/vmcontext/testing.go

```diff
@@ -53,9 +57,15 @@ func (f *Factories) NewKeyManager() vstate.KeyManager {
 	return newKeyManager()
 }
 
+type fakeRandSrc struct {
+}
+
+func (r fakeRandSrc) Randomness(_ context.Context, _ acrypto.DomainSeparationTag, _ abi.ChainEpoch, _ []byte) (abi.Randomness, error) {
+	panic("implement me")
+}
+
 func (f *Factories) NewRandomnessSource() vstate.RandomnessSource {
-	// TODO implement in a follow on
-	return nil
+	return &fakeRandSrc{}
```

Comment:
I have something coming soon. But I think for chain validation tests, we'll need this to be provided by the chain val code as part of the fake execution context that it provides. Otherwise different implementations will fake this differently.